packages/flutter/lib/src/material/pickers/calendar_date_picker.dart

```diff
@@ -0,0 +1,1050 @@
+        );
+    }
+    return null;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final Widget subheader = _buildSubheader();
+    final Widget picker = Flexible(
+      child: SingleChildScrollView(
+        child: SizedBox(
+          height: _maxDayPickerHeight,
+          child: _buildPicker(),
+        ),
+      ),
+    );
+
+    final Widget dialog = OrientationBuilder(
+      builder: (BuildContext context, Orientation orientation) {
+        assert(orientation != null);
+        switch (orientation) {
+          case Orientation.portrait:
+            return Column(
+              mainAxisSize: MainAxisSize.min,
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: <Widget>[
+                subheader,
+                picker,
+              ],
+            );
+          case Orientation.landscape:
+            return Column(
+              mainAxisSize: MainAxisSize.min,
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: <Widget>[
+                subheader,
+                picker,
+              ],
+            );
+        }
+        return null;
+      });
+
+    return dialog;
+  }
+
+  @override
+  void dispose() {
+    _pageController?.dispose();
+    super.dispose();
+  }
+}
+
+/// The subheader for the [DatePickerDialog].
+///
+/// This appears above the calendar grid and allows the user to toggle the
+/// [DatePickerMode] and quickly switch between next/previous months.
+class _DatePickerSubheader extends StatefulWidget {
+  const _DatePickerSubheader({
+    @required this.mode,
+    @required this.nextTooltipText,
+    @required this.previousTooltipText,
+    @required this.title,
+    @required this.onNextPressed,
+    @required this.onPreviousPressed,
+    @required this.onTitlePressed,
+  });
+
+  /// The current display of the calendar picker.
+  final DatePickerMode mode;
+
+  /// The tooltip text for the next month button.
+  final String nextTooltipText;
+
+  /// The tooltip text for the previous month button.
+  final String previousTooltipText;
+
+  /// The text that displays the current month/year being viewed.
+  final String title;
+
+  /// The callback when the next month button is pressed.
+  final VoidCallback onNextPressed;
+
+  /// The callback when the previous month button is pressed.
+  final VoidCallback onPreviousPressed;
+
+  /// The callback when the title is pressed.
+  final VoidCallback onTitlePressed;
+
+  @override
+  _DatePickerSubheaderState createState() => _DatePickerSubheaderState();
+}
+
+class _DatePickerSubheaderState extends State<_DatePickerSubheader>
+    with SingleTickerProviderStateMixin {
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = AnimationController(
+      value: widget.mode == DatePickerMode.year ? 1 : 0,
+      duration: const Duration(milliseconds: 200),
+      vsync: this,
+    )..addListener(() => setState(() {
```

Comment:
Typically this is done by wrapping the `build()` widget with an AnimatedBuilder whose animation is _controller. In this case it looks like only the Transform widget needs to be wrapped with an AnimatedBuilder .. in fact maybe a RotationTransition would be better than an AnimatedBuilder